tests/testthat/test-sparsevctrs.R

```diff
@@ -53,3 +59,67 @@ test_that("sparse tibble can be passed to `fit() - xy", {
 
   expect_no_error(wf_fit <- fit(wf_spec, hotel_data))
 })
+
+test_that("sparse matrices can be passed to `fit() - recipe", {
+  skip_if_not_installed("glmnet")
+  # Make materialization of sparse vectors throw an message
+  # https://r-lib.github.io/sparsevctrs/dev/reference/sparsevctrs_options.html
+  withr::local_options("sparsevctrs.verbose_materialize" = 1)
+
+  hotel_data <- sparse_hotel_rates()
+
+  spec <- parsnip::linear_reg(penalty = 0) %>%
+    parsnip::set_mode("regression") %>%
+    parsnip::set_engine("glmnet")
+
+  rec <- recipes::recipe(avg_price_per_room ~ ., data = hotel_data)
+
+  wf_spec <- workflow() %>%
+    add_recipe(rec) %>%
+    add_model(spec)
+
+  # We expect 1 materialization - the outcome
+  expect_snapshot(wf_fit <- fit(wf_spec, hotel_data))
```

Comment:
so we can't avoid materialization completely. Because the outcome is sparse. So instead we throw a message when materialization happens. And we just manually have to check that it only shows up once

Reply:
which is embedded in the snapshot